File: searchingAlgorithms/linearSearch.py
def linear_search(arr, target):
    """
    Function to perform linear search on a list.

    :param arr: List of elements
    :param target: The element to search for
    :return: Index of the target element if found, otherwise -1
    """
    for i in range(len(arr)):
        if arr[i] == target:
            return i  # Return the index if the element is found
    return -1  # Return -1 if the element is not found


# Example usage:
arr = [10, 23, 45, 70, 11, 15]
target = 70
result = linear_search(arr, target)

if result != -1:
    print(f"Element found at index {result}")
else:
    print("Element not found")

File: searchingAlgorithms/ LinearSearch.java
import java.util.List;

public class LinearSearch {
    public static int linearSearch(List<Integer> arr, int target) {
        for (int i = 0; i < arr.size(); i++) {
            if (arr.get(i) == target) {
                return i; // Return the index if the element is found
            }
        }
        return -1; // Return -1 if the element is not found
    }

    public static void main(String[] args) {
        List<Integer> arr = List.of(10, 23, 45, 70, 11, 15);
        int target = 70;
        int result = linearSearch(arr, target);

        if (result != -1) {
            System.out.println("Element found at index " + result);
        } else {
            System.out.println("Element not found");
        }
    }
}

